src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let myString = '';
+
+    if (this.length < 1) {
+      return myString;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      // myString = myString + separator;
+
+      if (this[i] === 'null') {
+        myString = myString + 'null';
+      } else if (this[i] === false && typeof (this[i]) === 'boolean') {
+        myString = myString + 'false';
+      } else if (this[i] !== null && this[i] !== undefined && this[i] !== '') {
```

Comment:
Avoid using else...if. For now your code looks complicated. 
Try using switch...case and/or ternary operator instead.